### ISSUE TITLE
fix: no salvage and subrogation modeling reduces net claim cost accuracy

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#1249]
## Problem Statement

The claim processing pipeline in `manufacturer_claims.py` and the loss distribution models in `loss_distributions.py` have **no salvage or subrogation recovery modeling**. All generated claims are treated as net of salvage/subrogation from inception, but no explicit recovery mechanism exists.

In manufacturing insurance, salvage (recovery from damaged goods) and subrogation (recovery from third parties at fault) can reduce ultimate claim costs by 5-15% for property lines and 10-25% for liability lines involving product defects. Per ASOP 43 (Unpaid Claim Estimates), the actuary should consider anticipated salvage and subrogation in loss reserve estimates.

Without explicit salvage/subrogation:
1. Loss distributions overstate net incurred losses
2. Reserve estimates are biased high (no offsetting recovery)
3. Insurance premium calculations based on gross losses overcharge

## v1.0 Impact

**Low-Medium** — Omitting salvage/subrogation systematically overstates loss costs and reserves. For manufacturing, product liability subrogation can be material (10%+ of incurred losses).

## Affected Code

- `ergodic_insurance/loss_distributions.py` — Loss generators produce gross losses with no recovery factor
- `ergodic_insurance/manufacturer_claims.py:238-438` — `process_insurance_claim()` has no salvage/subrogation offset
- `ergodic_insurance/claim_liability.py` — `ClaimLiability` tracks remaining without salvage receivable
- `ergodic_insurance/claim_development.py` — Development patterns assume net of S&S but no explicit modeling

## Current Behavior

```python
# Loss generated at $500K — no recovery applied
loss = LossEvent(amount=500_000, category="large")
company_payment, insurance_payment = mfr.process_insurance_claim(500_000, deductible=50_000)
# company_payment = $50K, insurance_payment = $450K
# No subrogation recovery ever reduces these amounts
```

## Expected Behavior

```python
# Loss generated at $500K with potential 15% subrogation recovery
loss = LossEvent(amount=500_000, category="large", subrogation_potential=0.15)
# Over time, recover ~$75K from third parties
# Net incurred = $425K after subrogation
```

## Recommended Approach

1. Add a `salvage_subrogation_ratio` parameter to `LossDistributionConfig` (default 0.0 for backward compatibility)
2. Add a recovery phase to `ClaimLiability` where a percentage of paid amounts are recovered over 1-3 years
3. `CashFlowProjector` should project salvage/subrogation recoveries as negative cash flows
4. Apply S&S ratios by line of business (property vs. liability)

## Acceptance Criteria

- [ ] `LossDistributionConfig` accepts salvage/subrogation ratio
- [ ] `ClaimLiability` tracks anticipated and actual recoveries
- [ ] Net incurred = gross incurred - S&S recoveries
- [ ] Default ratio of 0.0 preserves current behavior
- [ ] Unit tests verify recovery reduces net ultimate
